### ISSUE TITLE
Go 1.19 support

### ISSUE THREAD

**Issue by q9f** [ChainSafe/gossamer#2768]
Hello 😊 

## Issue summary

- compiling gossamer should work with the latest go version
- however, the dependency https://github.com/lucas-clemente/quic-go does not support 1.19 yet

```
~/.opt/ChainSafe/gossamer development 12s
❯ go version
go version go1.19 linux/amd64
```

## Other information and links

```
# github.com/lucas-clemente/quic-go/internal/qtls
../../../.go/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.1/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
make: *** [Makefile:73: build] Error 2
```

How are you managing go versions usually?

**Comment by q9f** [comment]
Just noticed #2719

Wondering if bumping quic to 0.28 would address this?

**Comment by qdm12** [comment]
This is blocked by a bunch of dependency upgrade regarding libp2p libraries, and substantial manual work.
Definitely to do for the fun of it, but definitely not a priority either. I have a local branch with a few changes (regarding deps upgrade, not Go 1.19 yet), I'll probably finish it in my 'friday evening refactoring fun' 😉

**Comment by qdm12** [comment]
Blocked by https://github.com/libp2p/go-libp2p-kad-dht/issues/790
